src/main/java/ma/hibernate/embeddable/dao/impl/NewsPostDaoImpl.java

```diff
@@ -11,11 +22,40 @@ public NewsPostDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public NewsPost save(NewsPost post) {
-        return null;
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Transaction transaction = null;
+        try (Session session = sessionFactory.openSession()) {
+            transaction = session.beginTransaction();
+            session.save(post);
+            transaction.commit();
+            return post;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("can't save post to db", e);
+        }
     }
 
     @Override
     public List<NewsPost> getWithMetadataSizeLargerThan(long size) {
-        return null;
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        try (Session session = sessionFactory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<NewsPost> criteriaQuery = criteriaBuilder.createQuery(NewsPost.class);
+            Root<NewsPost> root = criteriaQuery.from(NewsPost.class);
+            criteriaQuery.select(root);
+            Query<NewsPost> query = session.createQuery(criteriaQuery);
```

Comment:
@EliosPride do it in chain (like stream API), no need to create so much variables for each method.